src/Npgsql/Internal/PgReader.cs

```diff
@@ -87,11 +87,17 @@ internal void Revert(int size, int startPos, Size bufferRequirement)
         _currentSize = size;
     }
 
-    [Conditional("DEBUG")]
     void CheckBounds(int count)
     {
-        if (count > FieldRemaining)
-            ThrowHelper.ThrowInvalidOperationException("Attempt to read past the end of the field.");
+        if (NpgsqlReadBuffer.BufferBoundsChecks)
+            Core(count);
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        void Core(int count)
```

Comment:
Should be OK to inline this given that the whole code is in the BufferBoundsChecks condition, no?

Reply:
Not for NativeAOT where we can't eliminate the code. 

It'll make the CheckBounds body larger which then might cause ReadInt etc to not be inlined.